libs/Onboarding/src/Onboarding/Accounts/Service.cpp

```diff
@@ -0,0 +1,438 @@
+{
+    auto response = StatusGo::Accounts::generateIdenticon(publicKey);
+    if(response.containsError())
+    {
+        qWarning() << response.error.message;
+        return QString();
+    }
+
+    return response.result;
+}
+
+//bool Service::verifyAccountPassword(const QString& account, const QString& password)
+//{
+//    // TODO:
+//    return false;
+//}
+
+DerivedAccounts Service::storeDerivedAccounts(const QString& accountId, const QString& hashedPassword,
+                                              const QVector<QString>& paths)
+{
+    auto response = StatusGo::Accounts::storeDerivedAccounts(accountId, hashedPassword, paths);
+    if(response.containsError())
+    {
+        qWarning() << response.error.message;
+        return DerivedAccounts();
+    }
+    return DerivedAccounts::toDerivedAccounts(response.result);
+}
+
+StoredAccountDto Service::storeAccount(const QString& accountId, const QString& hashedPassword)
+{
+    auto response = StatusGo::Accounts::storeAccount(accountId, hashedPassword);
+    if(response.containsError())
+    {
+        qWarning() << response.error.message;
+        return StoredAccountDto();
+    }
+    return toStoredAccountDto(response.result);
+}
+
+AccountDto Service::saveAccountAndLogin(const QString& hashedPassword, const QJsonObject& account,
+                                        const QJsonArray& subaccounts, const QJsonObject& settings,
+                                        const QJsonObject& config)
+{
+    if(!StatusGo::Accounts::saveAccountAndLogin(hashedPassword, account, subaccounts, settings, config)) {
+        qWarning() << "Failed saving acccount" << account.value("name");
+        return AccountDto();
+    }
+
+    m_isFirstTimeAccountLogin = true;
+    return AccountDto::toAccountDto(account);
+}
+
+QJsonObject Service::prepareAccountJsonObject(const GeneratedAccountDto& account, const QString &displayName) const
+{
+    return QJsonObject{{"name", displayName.isEmpty() ? account.alias : displayName},
+        {"address", account.address},
+        {"photo-path", account.identicon},
+        {"identicon", account.identicon},
+        {"key-uid", account.keyUid},
+        {"keycard-pairing", QJsonValue()}};
+}
+
+QJsonObject Service::getAccountDataForAccountId(const QString &accountId, const QString &displayName) const
+{
+
+    foreach(const GeneratedAccountDto& acc, m_generatedAccounts)
+    {
+        if(acc.id == accountId)
+        {
+            return Service::prepareAccountJsonObject(acc, displayName);
+        }
+    }
+
+    if(m_importedAccount.isValid())
+    {
+        if(m_importedAccount.id == accountId)
+        {
+            return Service::prepareAccountJsonObject(m_importedAccount, displayName);
+        }
+    }
+
+    qDebug() << "account not found";
+    return QJsonObject();
+}
+
+QJsonArray Service::prepareSubaccountJsonObject(const GeneratedAccountDto& account, const QString &displayName) const
+{
+    return QJsonArray{
```

Comment:
I believe you can directly return an `std::initializer_list` (braced syntax) from here